ios/authentication.md

```diff
@@ -785,3 +819,402 @@ Note : `AWSGoogleSignIn` is only needed for using Google Login in your app and `
+{: .callout .callout--info}
+
+*Authorization code grant* is the recommended choice for security reasons.
+{: .callout .callout--info} 
+
+- Choose item(s) from *OAuth Scopes*.
+
+Note: `openid` is required for `phone`, `email` or `profile`. Also `openid` is required to get the id token from the Cognito authorization server.
+{: .callout .callout--info}
+
+- Click 'Save Changes'. 
+
+#### Setup Amazon Cognito Hosted UI in iOS App
+
+1. To configure your application for hosted UI, you need to use *HostedUI* options. Update your `awsconfiguration.json` file to add a new configuration for `MobileClient`. The configuration should look like this:
+
+    ```json
+    {
+        "IdentityManager": {
+            ...
+        },
+        "CredentialsProvider": {
+            ...
+        },
+        "CognitoUserPool": {
+            ...
+        },
+        "MobileClient": {
+            "Default": {
+                "HostedUI": {
+                    "WebDomain": "https://YOUR_AUTH_DOMAIN.auth.us-west-2.amazoncognito.com",
+                    "AppClientId": "YOUR_APP_CLIENT_ID",
+                    "AppClientSecret": "YOUR_APP_CLIENT_SECRET",
+                    "SignInRedirectURI": "myapp://",
+                    "SignOutRedirectURI": "myapp://",
+                    "Scopes": ["openid", "email"]
+                }
+            }
+        }
+    }
+    ```
+
+1. Add `myapp://` to your app's URL schemes:
+
+    Right-click Info.plist and then choose Open As > Source Code.
+
+    Add the following entry in URL scheme:
+
+    ```xml
+        <plist version="1.0">
+
+        <dict>
+        <!-- YOUR OTHER PLIST ENTRIES HERE -->
+
+        <!-- ADD AN ENTRY TO CFBundleURLTypes for Cognito Auth -->
+        <!-- IF YOU DO NOT HAVE CFBundleURLTypes, YOU CAN COPY THE WHOLE BLOCK BELOW -->
+        <key>CFBundleURLTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleURLSchemes</key>
+                <array>
+                    <string>myapp</string>
+                </array>
+            </dict>
+        </array>
+
+        <!-- ... -->
+        </dict>
+    ```
+
+#### Launching the Hosted UI
+
+To launch the Hosted UI from from your application, you can use the `showSignIn` API of `AWSMobileClient.sharedInstance()`:
+
+```swift
+// Optionally override the scopes based on the usecase.
+let hostedUIOptions = HostedUIOptions(scopes: ["openid", "email"])
+
+// Present the Hosted UI sign in.
+AWSMobileClient.sharedInstance().showSignIn(navigationController: self.navigationController!, hostedUIOptions: hostedUIOptions) { (userState, error) in
+    if let error = error as? AWSMobileClientError {
+        print(error.localizedDescription)
+    }
+    if let userState = userState {
+        print("Status: \(userState.rawValue)")
+    }
+}
+```
+
+> Optional: If your app deployment target is < iOS 11, add the following callback in your App Delegate's `application:open url` method:
```

Comment:
Is this related to SFSafariViewController vs. SFAuthenticationSession? Or is there a reason why they might use this option otherwise? Either way can we clarify in one sentence?

Reply:
Done

Reply:
> Optional: If your app deployment target is < iOS 11, add the following callback in your App Delegate's `application:open url` method. This callback is required, because `SFSafariViewController` is used in < iOS 11 to integrate with the Hosted UI.